File: adventofcode/java/year2016/day04/Day04Test.java
package adventofcode.java.year2016.day04;

import java.io.FileReader;
import java.io.IOException;

import org.junit.Test;
import org.junit.Assert;

public class Day04Test {
    @Test
    public void part1Example() throws IOException {
        try (var input = new FileReader("adventofcode/java/testdata/year2016/day04/example")) {
            var output = "1514";
            var result = Day04.part1(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }

    @Test
    public void part1Actual() throws IOException {
        try (var input = new FileReader("adventofcode/data/year2016/day04/actual.in")) {
            var output = "361724";
            var result = Day04.part1(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }

    @Test
    public void part2Actual() throws IOException {
        try (var input = new FileReader("adventofcode/data/year2016/day04/actual.in")) {
            var output = "482";
            var result = Day04.part2(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }
}

File: adventofcode/cpp/year2019/day04/day04_test.cc
#include "adventofcode/cpp/year2019/day04/day04.h"

#include <fstream>

#include "gtest/gtest.h"

#include "adventofcode/cpp/adventofcode.h"

TEST(Year2019Day04, Part1Actual) {
  std::ifstream input("adventofcode/data/year2019/day04/actual.in");
  std::string output = "1246";
  adventofcode::answer_t a = day04::part1(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}

TEST(Year2019Day04, Part2Actual) {
  std::ifstream input("adventofcode/data/year2019/day04/actual.in");
  std::string output = "814";
  adventofcode::answer_t a = day04::part2(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}
